### DISCUSSION TITLE
Do we have the import.meta.env.SSR

### DISCUSSION THREAD

**Discussion by nhattran998** [web-infra-dev/rsbuild#3623] [Q&A] [answered]
Thank teams for the great tools <3.

After reading the docs - https://rsbuild.dev/guide/advanced/env-vars and https://rsbuild.dev/guide/advanced/ssr, I don't see any place disscuss about this. So, how can I do this with RsBuild 🙏.

**Comment by chenjiahan** [comment, marked as answer]
Rsbuild injects the following environment variables by default:

- `import.meta.env.MODE`
- `import.meta.env.BASE_URL`
- `import.meta.env.PROD`
- `import.meta.env.DEV`

For `import.meta.env.SSR`, you can set it through the `environments` configuration option:

```ts title="rsbuild.config.ts"
export default defineConfig({
  environments: {
    web: {
      source: {
        define: {
          'import.meta.env.SSR': false,
        },
      },
    },
    node: {
      source: {
        define: {
          'import.meta.env.SSR': true,
        },
      },
      output: {
        target: 'node',
      },
    },
  },
});
```

**Reply by nhattran998** [reply]
Thank for your quick response 🙏

**Comment by SoonIter** [comment]
Update: `import.meta.env.SSR` is now supported as a built-in Rsbuild env value since v2.0.7, via https://github.com/web-infra-dev/rsbuild/pull/7700 (merged on 2026-05-20).

It is `true` for Node target builds and `false` for other targets, so after upgrading to v2.0.7 or later, you no longer need to manually define `import.meta.env.SSR` through `source.define` for the common web/node setup.
